app/src/main/java/backgammon/Move.java

```diff
@@ -0,0 +1,63 @@
+package backgammon;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Move {
+
+  private final int from;
+  private final int to;
+  private final List<Movelet> movelets = new ArrayList<>();
+
+
+  public Move(int from, int to, List<Integer> dice, boolean isDoubles) {
+    this.from = from;
+    this.to = to;
+
+    int die = dice.get(0);
+
+    for (int i = 0; i < dice.size() && isDoubles; i++) {
+      movelets.add(new Movelet(from - die * i, from - die * (i + 1), die ));
+    }
+
+    if (!isDoubles) {
+      movelets.add(new Movelet(from, from - die, die ));
+      if (dice.size() == 2)
+        movelets.add(new Movelet(from - die, from - die - dice.get(1), dice.get(1)));
```

Comment:
Look at these two lines here, you're needing to reuse this expression `from - die` numerous times.
Notice that you're needing to use this value regardless of what branch you're in, and even before then in the loop.
Please by all means make an actual variable if you'll even use this more than twice in separate instances just so that it more easily show your logic instead of burying it in these statements.

Reply:
In this case `from - die` has a different purpose each time it was used so I never assigned a variable to it. On the top line it is the integer corresponding to the destination point on the board, while the bottom it is the origin point.